runtime/compiler/env/j9methodServer.cpp

```diff
@@ -166,8 +166,22 @@ TR_ResolvedJ9JITaaSServerMethod::classOfStatic(I_32 cpIndex, bool returnClassFor
    // if method is unresolved, return right away
    if (cpIndex < 0)
       return nullptr;
+
+   auto &classOfStaticCache = getJ9ClassInfo(_fe->_compInfoPT, _ramClass)._classOfStaticCache;
+   if (!classOfStaticCache)
+      {
+      // initialize cache, called once per ram class
+      classOfStaticCache = new (PERSISTENT_NEW) PersistentUnorderedMap<int32_t, TR_OpaqueClassBlock *>(PersistentUnorderedMap<int32_t, TR_OpaqueClassBlock *>::allocator_type(TR::Compiler->persistentAllocator()));
+      }
+   
+   auto it = classOfStaticCache->find(cpIndex);
+   if (it != classOfStaticCache->end())
+      return it->second;
+
    _stream->write(JITaaS::J9ServerMessageType::ResolvedMethod_classOfStatic, _remoteMirror, cpIndex, returnClassForAOT);
-   return std::get<0>(_stream->read<TR_OpaqueClassBlock *>());
+   TR_OpaqueClassBlock *classOfStatic = std::get<0>(_stream->read<TR_OpaqueClassBlock *>());
+   classOfStaticCache->insert({cpIndex, classOfStatic});
```

Comment:
Caching should take place only if the client returns a valid class. It's possible that at this time the cp entry is not resolved or that the class is not yet fully initialized, but things may change in the future.